### DISCUSSION TITLE
Project status

### DISCUSSION THREAD

**Discussion by klaypigeon** [mickem/nscp#767] [General]
Is this project still alive? Looks like latest Stable is from 2018. I see a bunch of outstanding PRs.

**Comment by mickem** [comment]
I have responded to this before. but lets re-iterate...

Unfortunately NSClient++ is something I do "for free" in my spare time which means I sometimes have time and other times not.
Before I had a number of sponsor of the project which allowed me to actually spend more time on this as I could "pay myself" (i.e. take time off work) for spending time on it.

Also last few years I have spent more of my my spare timer with my two kids and family which has negatively impacted the time I have to spend in development of NSClient++.
That said I have for the last 2 years spent countless hours trying to get a working build environment up and running with modern versions of all libs. This is unfortunately not a simple task as the project has many dependencies and C++ unfortunately a language which makes dependency management much harder then it has to be...

That said I have recently decided to set this up with Github Actions instead of complicated local scripts and wrappers and what not.
It will however take some time to get this up and running... So I cannot promise when it will be done.

Currently I have builds working on Linux (as this is bound to be easiest) and am currently working on moving from python 2 -> 3.
Hopefully I will be able to fix this soon and after which I can start on the Windows builds.
After which IO also plan to start removing some third party dependencies to make things simpler...

Feel free to help out, as that any help would speed this along.

Once I have a working build environment development will resume at a much speedier pace...

But ultimately this, and many other open source projects, suffer from lack of funding so I have to squeeze out what time I can...

// Michael Medin

**Reply by klaypigeon** [reply]
Thanks for the Reply.  Glad it's still on the radar. It's too bad all the Enterprises out there that are distributing your code with their systems or installing it are not being better partners to you. Maybe you need a marketing partner to get out there and bang on some doors and get folks to cough up some subscription bucks. Its not like people are running this on their home machines, this is all corporate usage.

**Reply by lausser** [reply]
Hi Michael,
on april 22 i wrote you an email offering to be payed at least for setting up a replicable build environment. Also, i have a customer who can definitely imagine paying you as an external contractor. So, no need to work for free on NSClient++. And, as the project got so much attention, as soon as it is possible to compile it (and i have seen even skilled people giving up on trying to do so), there are volunteers for the necessary updates on ssl/ciphers. Please contact me if this is an option for you.
Gerhard

**Reply by mickem** [reply]
A reproducible build environment is what I am setting up here: https://github.com/mickem/nscp/pull/763 So that is definitely in the works. As you can see in the diff there was considerable work to get all libs reasonably updated, I was doing good progress in December and had hopes to finished it by year end but alas I had some family illness that I had to take care of since late December...  But with some luck it will be done in a week or so... (with the exception of dot-net and all modules requiring that). As I have everything working locally and just need to reproduce that with github actions...

**Reply by lausser** [reply]
That's good news! Thank you, Michael, for the answer. If you need resources, funding, whatever makes it easy for you, don't hesitate to drop me a mail. 
I can assure you that (technical issues aside) over the past years, many people worried about whether everything is okay with you and your family. Take care.

**Reply by lausser** [reply]
Sorry, if i'm annoying. Do you see any chance for an update in the near future? At least for a working build process?

**Reply by lausser** [reply]
Well, it seems nsclient is abandoned now and you are dedicating your time and attention to more important things...
Let's pass the torch to
https://github.com/ConSol-Monitoring/snclient

**Reply by firatalkis** [reply]
@lausser, thanks for you help in this issue. I will try "snclient+ agent" at the soonest time

**Reply by delboy1966** [reply]
How do you install snclient on windows?

**Reply by sni** [reply]
could we move discussions and questions about snclient to the designated discussions page -> https://github.com/ConSol-Monitoring/snclient/discussions

**Comment by klaypigeon** [comment]
> Well, it seems nsclient is abandoned now and you are dedicating your time and attention to more important things... Let's pass the torch to https://github.com/ConSol-Monitoring/snclient

Thanks @lausser , I am monitoring your progress on this. I have been impressed with some of ConSols work, and it would be nice to see this under more active development. Are you sure you are ready for the support burden this may present though?

**Reply by delboy1966** [reply]
Who says its abandoned.

**Comment by lausser** [comment]
First and foremost, the true credit goes to @sni, who founded the snclient project. As for the support burden... there will be no support.
We have customers who have no alternative but to uninstall the NSClient++ agents because they have been (since years) under pressure from their security teams and are unwilling to wait any longer for a potential future update. These customers have sponsored the development of snclient thus far and, due to their support contracts, receive any assistance they need.
Everyone else is welcome to utilize the software, but it should be noted that open source does not imply free consulting.

**Reply by klaypigeon** [reply]
Maybe a community support forum or something will pop up. Always nice when we can help each other when we get stuck writing a complicated check or can't figure the syntax or don't know how to check XYZ.

**Reply by sni** [reply]
i enabled github discussions for snclient

**Comment by klaypigeon** [comment]
For those that are watching, Opsview, which is now owned by ITRS is also releasing an infrastructure agent for mostly the same reasons, which is in Beta now. It appears to be all Python while Consul's effort is written in Go. Benefits to both I suppose. Anyway, just putting it out there.
https://github.com/ITRS-Group/infrastructure-agent
https://knowledge.opsview.com/docs/opsview-infrastructure-agent-beta-installation-and-operation

**Comment by firatalkis** [comment]
Thanks for this info. As soon as I will try it.

Sent from my iPhone

On 1 Jul 2023, at 00:28, klaypigeon ***@***.***> wrote:

﻿

For those that are watching, Opsview, which is now owned by ITRS is also releasing an infrastructure agent for mostly the same reasons, which is in Beta now. It appears to be all Python while Consul's effort is written in Go. Benefits to both I suppose. Anyway, just putting it out there.
https://github.com/ITRS-Group/infrastructure-agent
https://knowledge.opsview.com/docs/opsview-infrastructure-agent-beta-installation-and-operation

**Reply by mickem** [reply]
A short update is that I have now completed the first pipeline for building NSClient++ on github actions. So expect a new version out in the next week or so.


Currently missing:
* Win32 (Should be simple)
* Installer (Should be simple)
* NSCA Support, A bit more involved as we need to build Crypto++ for this
* Dot net modules, not sure how hard this will be but very few are using this so not sure it is critical.


A question for the community: What would be most valuable going forward:

1. Dot net modules
2. Updates to support for "modern windows"
3. New NRPE support
4. A REST stand-alone binary so you can use rest instead of NRPE (from linux/*)

As we now have a working build you can expect a lot more updates going forward... And sorry for this taking a lot longer but as I had said before I have had some health issues in the close family that have been taking up all my time...

**Reply by ICONsa-GVA** [reply]
"New NRPE support" to assure compatibility with NRPE v4

**Comment by mickem** [comment]
If anyone wants to try out the new builds you can find one [here](https://github.com/mickem/nscp/actions/runs/5707689351). But in the next few days I will merge this branch and then they will be under releases as well...

**Reply by ICONsa-GVA** [reply]
I'm interesting to try new build but you miss some dll in zip file. --> boost_* and libprotobuf.dll

**Reply by mickem** [reply]
Sorry for the late reply, you are correct. New builds have correct DLLs.

I am currently working on the installer after which I will post actual builds which will make it easier to try the new builds...
This will hopefully be done soon...

Link to last build here: https://github.com/mickem/nscp/actions/runs/6816142367 (or click the actions tab and pick latest build from there).

**Comment by mickem** [comment]
I will close this as we have regular new versions out now...
